test_fms/fms2_io/test_io_simple.F90

```diff
@@ -69,22 +105,75 @@ program test_io_simple
 
   call netcdf_io_init(ncchksz, header_buffer_val, netcdf_default_format)
 
-  ! Open a netCDF file and initialize the file object.
-  call open_check(open_file(fileobj, "test_io_simple.nc", "overwrite", &
-       domain, nc_format="64bit", is_restart=.false.))
+  do i = 1, 3
+     write(testfile,'(a,a,a)') 'test_io_simple_', trim(my_format(i)), '.nc'
+     
+     ! Open a netCDF file and initialize the file object.
+     call open_check(open_file(fileobj, testfile, "overwrite", &
+          domain, nc_format=my_format(1), is_restart=.false.))
 
-  ! Close the file.
-  call close_file(fileobj)
+     ! Add a global attribute.
+     call register_global_attribute(fileobj, "globalatt1", real(7., kind=real64))
 
-  ! Check for expected netcdf file.
-  if (mpp_pe() .eq. mpp_root_pe()) then
-     err = nf90_open('test_io_simple.tile1.nc', nf90_nowrite, ncid)
-     if (err .ne. NF90_NOERR) stop 1
-     err = nf90_close(ncid)
-     if (err .ne. NF90_NOERR) stop 90
-  endif
+     ! Add a dimension.
+     call register_axis(fileobj, "lon", "x")
+
+     ! Add a coordinate variable for the dimension.
+     names(1) = "lon"
+     call register_field(fileobj, "lon", "double", names(1:1))
+     call write_data(fileobj, "lon", double_buffer)     
+          
+     ! Close the file.
+     call close_file(fileobj)
+     
+     call mpi_barrier(mpi_comm_world, err)
+     call mpi_check(err)
+
+     ! Check for expected netcdf file.
+     if (mpp_pe() .eq. mpp_root_pe()) then
+        write(testfile,'(a,a,a)') 'test_io_simple_', trim(my_format(i)), '.tile1.nc'
+        err = nf90_open(testfile, nf90_nowrite, ncid)
```

Comment:
Perhaps you should check if the file_type is correct?

Reply:
Thanks! Good catch! I will fix...

Reply:
OK, turns out that the command to test the file type is not present in the netCDF F90 API! Whoops! This is a mistake I made about 15 years ago. ;-)

Well, I have added a ticket to the netcdf-fortran project, and I will hopefully get around to adding the nf90_inq_format() function soon, but it won't be available until the next release of netcdf-fortran anyway, so we don't want to use it in the test yet. So we will just skip testing the format for now.